pulp_2to3_migration/app/plugin/deb/pulp_2to3_models.py

```diff
@@ -184,3 +184,104 @@ def create_pulp3_content(self):
+        pulp2_ids = pulp2_unit_map.keys()
+        pulp2_unit_batch = pulp2_models.DebRelease.objects.filter(id__in=pulp2_ids)
+        units_to_save = [Pulp2DebRelease(distribution=release.distribution,
+                                         codename=release.codename,
+                                         suite=release.suite,
+                                         pulp2content=pulp2_unit_map[release.id],)
+                         for release in pulp2_unit_batch]
+
+        cls.objects.bulk_create(
+            units_to_save,
+            ignore_conflicts=True,
+            batch_size=DEFAULT_BATCH_SIZE
+        )
+
+    def create_pulp3_content(self):
+        """
+        Create a Pulp 3 detail Content unit for saving it later in a bulk operation.
+
+        Return an unsaved Pulp 3 Content
+        """
+        pulp3_release = pulp3_models.Release(
+            distribution=self.distribution.strip("/"),
+            codename=self.codename,
+            suite=self.suite,
+        )
+
+        return (pulp3_release, None)
+
+
+class Pulp2DebComponent(Pulp2to3Content):
+    """
+    Pulp 2to3 detail content model to store pulp 2 DebComponent content details for Pulp 3
+    content creation.
+    """
+    pulp2_type = 'deb_component'
+    distribution = django_models.TextField()
+    codename = django_models.TextField()
+    component = django_models.TextField()
+
+    class Meta:
+        default_related_name = 'deb_component_detail_model'
+        # TODO: These are probably not the uniqueness constraints I want...
+        unique_together = ('component', 'distribution')
+
+    @classmethod
+    def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 content with all the fields needed to create a Pulp 3 Content
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+        """
+        pulp2_unit_map = {pulp2unit.pulp2_id: pulp2unit for pulp2unit in content_batch}
+        pulp2_ids = pulp2_unit_map.keys()
+        pulp2_unit_batch = pulp2_models.DebComponent.objects.filter(id__in=pulp2_ids)
+        units_to_save = [Pulp2DebComponent(distribution=component.distribution,
+                                           codename=component.release,
+                                           component=component.name,
+                                           pulp2content=pulp2_unit_map[component.id],)
+                         for component in pulp2_unit_batch]
+
+        cls.objects.bulk_create(
+            units_to_save,
+            ignore_conflicts=True,
+            batch_size=DEFAULT_BATCH_SIZE
+        )
+
+    def create_pulp3_content(self):
+        """
+        Create a Pulp 3 detail Content unit for saving it later in a bulk operation.
+
+        Return an unsaved Pulp 3 Content
+        """
+        release_relation = {'release': '???'}
+        return (pulp3_models.ReleaseComponent(component=self.component), release_relation)
```

Comment:
I opened this Draft PR, because I need some help:
The Pulp3 `ReleaseComponent` I am creating here, has the field `release` of type `ForeignKey` that should point at a Pulp3 content unit of type `Release`. I suspect that this relation should be returned to the calling instance via what I have here called `release_relation`. However, I have no clear understanding of what `release_relation` needs to contain for the mechanism to function as designed. I want a reference to the `Release` component unit, which will end up in the same repository version as the `ReleaseComponent` I am creating, and also has the same value in it's `distribution` field, as `self.distribution`. Any help is appreciated!

Reply:
Extra Information: If I run the code as is (with `release_relation = {'release': '???'}`) I get the following error:
```
pulp_deb.app.models.content.ReleaseComponent.DoesNotExist: ReleaseComponent matching query does not exist.
```

Reply:
Ok, I found something that will allow me to continue for now. The docker migration pipeline has a special stage that picks up on the things that are returned using what I have called `release_relation` and processes them. I will try to emulate that mechanism.

Reply:
@quba42 , you should try to reuse the stages from your sync pipeline.
If you have this relation, I guess the sync pipeline has a stage or some code that deals with it, creates this relation.

Reply:
In the Debian sync pipeline, this is done right in the first stage, by ensuring one of the two related partners is pushed all the way through the pipeline right away, and is then immediately available for relating it to the other content type. That wont work for the migration pipeline, since the migration pipeline loops over each content type in turn and does not have both of them available at the same time at the start. However, I think I can emulate what docker is doing and retrieve the "to be associated" unit in a later dedicated `InterrelateContent` stage.

Will take me a while to implement though, especially since other priorities are intruding on my time again...